doc/developer/design/20250515_query_lifecycle_extensions.md

```diff
@@ -0,0 +1,125 @@
+# Extending the Query Lifecycle Tables
+
+- Associated: [#9140](https://github.com/MaterializeInc/database-issues/issues/9140)
```

Comment:
Can you also link the original [statement logging doc](https://github.com/MaterializeInc/materialize/blob/4487749289f1591e7918ebaa1458e23b5e40633f/doc/developer/design/20230519_statement_logging.md) and the [query lifecycle doc](https://github.com/MaterializeInc/materialize/blob/4487749289f1591e7918ebaa1458e23b5e40633f/doc/developer/design/20231204_query_lifecycle_events_logging.md)?

Reply:
Done